discovery_engine_core/web-api2/src/server.rs

```diff
@@ -77,11 +78,21 @@ where
         let addr = config.bind_address();
         init_tracing(config.log_file());
         let app_state = web::Data::new(A::AppState::try_from(config)?);
+        let json_config = web::JsonConfig::default()
+            .limit(524288)
+            .content_type(|mime| mime == mime::TEXT_PLAIN)
+            .error_handler(|err, _req| {
+                InternalError::from_response(err, HttpResponse::Conflict().finish()).into()
+            });
```

Comment:
I'm not sure why you also added this lines from the example.

Given the documentation of actix:

- `content_type` allows you to configure content types we do accept as json bodies, we don't have legacy client sending the `text/plain` as content type for json. So we don't need this.

- `error_handler` allows you do customize the error handling for json parsing failure. We don't need this (we have a middleware doing this uniformly for all "simple" errors)

So basically we just need `JsonConfig::default().limit(....);`.